libsupport/src/ArrowInterchange.cpp

```diff
@@ -70,31 +70,33 @@ ApproxArrayDataMemUse(const std::shared_ptr<arrow::ArrayData>& data) {
   return total_mem_use;
 }
 
-}  // anonymous namespace
-
 katana::Result<std::shared_ptr<arrow::Array>>
-katana::Unchunk(const std::shared_ptr<arrow::ChunkedArray>& original) {
+Unchunk(const std::shared_ptr<arrow::ChunkedArray>& original) {
   std::shared_ptr<arrow::Array> indices = KATANA_CHECKED(Indices(original));
   std::shared_ptr<arrow::ChunkedArray> chunked =
       KATANA_CHECKED(IndexedTake(original, indices));
   return chunked->chunk(0);
 }
 
-katana::Result<std::shared_ptr<arrow::ChunkedArray>>
-katana::Shuffle(const std::shared_ptr<arrow::ChunkedArray>& original) {
-  int64_t length = original->length();
-  // Build indices array, reusable across properties
-  std::vector<uint64_t> indices_vec(length);
-  // fills the vector from 0 to indices_vec.size()-1
-  std::iota(indices_vec.begin(), indices_vec.end(), 0);
-  std::shuffle(indices_vec.begin(), indices_vec.end(), katana::GetGenerator());
-  arrow::CTypeTraits<int64_t>::BuilderType builder;
-  KATANA_CHECKED(builder.Reserve(length));
-  for (int64_t i = 0; i < length; ++i) {
-    builder.UnsafeAppend(indices_vec[i]);
+}  // anonymous namespace
+
+katana::ErrorCode
+katana::ArrowToKatana(arrow::StatusCode code) {
+  KATANA_LOG_DEBUG_ASSERT(code != arrow::StatusCode::OK);
+
+  switch (code) {
+  case arrow::StatusCode::Invalid:
+    return ErrorCode::InvalidArgument;
+  case arrow::StatusCode::TypeError:
+    return ErrorCode::TypeError;
+  case arrow::StatusCode::AlreadyExists:
+    return ErrorCode::AlreadyExists;
+  case arrow::StatusCode::KeyError:
+  case arrow::StatusCode::IndexError:
+    return ErrorCode::NotFound;
+  default:
+    return ErrorCode::ArrowError;
```

Comment:
This function was moved to Result.h and should no longer be necessary here

Reply:
Thanks for weighing in.  I should have explicitly asked you about this.

What I find confusing is that `ArrowStatusCodeToKatana `is in Result.h, yet files like UpdateGraphView.cpp are calling `ArrowToKatana`.  That was why I left it in, even though it seemed extraneous. When I eliminate `ArrowToKatana`, I get copious build errors.

Reply:
Ok, there might have been concurrent modifications. Uses of `ArrowToKatana` can be replaced with `internal::ArrowStatusCodeToKatana` for now, and should probably eventually migrate to `KATANA_CHECKED`.
Or if you don't want to do that here, that's ok too

Reply:
You meant in the enterprise repo, got it. I'll make that change in https://github.com/KatanaGraph/katana-enterprise/pull/1415